### DISCUSSION TITLE
Why on Linux only(!) Worker::runAll PARSE COMMAND LINE PARAMS?!

### DISCUSSION THREAD

**Discussion by AlexFirelord** [walkor/Workerman#1022] [General]
protected static function parseCommand()
    {
        if (static::$_OS !== \OS_TYPE_LINUX) {
            return;
        }
...
        $usage = "Usage: php yourfile <command> [mode]...";
...
        if (!$command) {
            exit($usage);
        }

What logic hides beneath that behavior I wonder???

**Comment by blogdaren** [comment]
First of all, workerman supports Windows and Linux platforms, but many of Workerman's core functions are limited under the Windows platform, that’s why workerman have the logic shown in the title, For example, the important PHP extensions **PCNTL** and **POSIX** which workerman relies on can only be used under Unix-Like platforms. The specific differences are explained in detail in the documentation [here](https://www.workerman.net/windows).

**Reply by AlexFirelord** [reply]
It's not that case, with all respect - under Linux for some reason lib stops handling API function - instead requires some command line params - like standalone application which is bizarre :) And this functionality is hidden deep withing the library methods and not documented (as far I can see). Although the lib itself is great - but this little "feature" looks quite weird. I had to dig into the source code and just comment the that check out - and it works perfect...
